concent_api/core/payments/bankster.py

```diff
@@ -246,14 +247,15 @@ def finalize_payment(deposit_claim: DepositClaim) -> Optional[str]:
         elif deposit_claim.concent_use_case == ConcentUseCase.ADDITIONAL_VERIFICATION:
             subtask = Subtask.objects.filter(subtask_id=deposit_claim.subtask_id).first()  # pylint: disable=no-member
             if subtask is not None:
-                if subtask.requestor.public_key == deposit_claim.payer_deposit_account.client.public_key:
+                task_to_compute = deserialize_message(subtask.task_to_compute.data.tobytes())
+                if task_to_compute.requestor_ethereum_address == deposit_claim.payer_deposit_account.ethereum_address:
                     ethereum_transaction = service.force_subtask_payment(  # pylint: disable=no-value-for-parameter
                         requestor_eth_address=deposit_claim.payer_deposit_account.ethereum_address,
                         provider_eth_address=deposit_claim.payee_ethereum_address,
                         value=deposit_claim.amount,
                         subtask_id=deposit_claim.subtask_id,
                     )
-                elif subtask.provider.public_key == deposit_claim.payer_deposit_account.client.public_key:
+                elif task_to_compute.provider_ethereum_address == deposit_claim.payer_deposit_account.ethereum_address:
                     ethereum_transaction = service.cover_additional_verification_cost(  # pylint: disable=no-value-for-parameter
                         provider_eth_address=deposit_claim.payer_deposit_account.ethereum_address,
```

Comment:
The `else` branch should have `assert False`. The claim is always supposed to refer to either requestor or provider as payer.

Reply:
:+1: done